File: src/main/java/miragefairy2019/mod/lib/IngredientFactoryOreIngredientComplex.java
package miragefairy2019.mod.lib;

import com.google.gson.JsonObject;
import net.minecraft.item.crafting.Ingredient;
import net.minecraft.util.JsonUtils;
import net.minecraftforge.common.crafting.IIngredientFactory;
import net.minecraftforge.common.crafting.JsonContext;

/**
 * 耐久が削れたクラフティングツールを鉱石辞書名にマッチさせるためのIngredient
 */
public class IngredientFactoryOreIngredientComplex implements IIngredientFactory {

    @Override
    public Ingredient parse(JsonContext context, JsonObject json) {
        return new OreIngredientComplex(JsonUtils.getString(json, "ore"));
    }

}

File: src/main/java/miragefairy2019/export/IngredientFactoryOreIngredientComplex.kt
package miragefairy2019.export

import com.google.gson.JsonObject
import miragefairy2019.libkt.OreIngredientComplex
import net.minecraft.util.JsonUtils
import net.minecraftforge.common.crafting.IIngredientFactory
import net.minecraftforge.common.crafting.JsonContext

// TODO move: ResourceMakerによりexportではなくなった
/**
 * 耐久が削れたクラフティングツールを鉱石辞書名にマッチさせるためのIngredient
 */
@Suppress("unused") // リフレクション経由で呼ばれる
class IngredientFactoryOreIngredientComplex : IIngredientFactory {
    override fun parse(context: JsonContext, json: JsonObject) = OreIngredientComplex(JsonUtils.getString(json, "ore"))
}
